crates/holochain_state/src/source_chain.rs

```diff
@@ -680,7 +680,11 @@ impl SourceChain {
                     // If this is a countersigning session we want to withhold
                     // publishing the ops until the session is successful.
                     if is_countersigning_session {
-                        set_withhold_publish(txn, op_hash)?;
+                        set_withhold_publish(txn, op_hash.clone())?;
+                    }
+
+                    if is_private_entry {
+                        set_private_entry_store_entry(txn, op_hash)?;
                     }
```

Comment:
This goes beyond the scope of this PR but I don't think it's great to just keep performing extra queries to modify the newly created DB rows when the correct values could just have just been set in the original INSERT query, is there a reason we do it this way?

Reply:
Because this information comes from the header. There's nothing in the dhtop light to determine if it has a private entry.

Reply:
Yes, but we have the header at the time we're inserting the op, so we could insert the op and set the appropriate fields at the same time as part of the insert, no?

Reply:
Yeh I guess we could with this. A few of the others calls are specific to authoring ops vs inserting them when you get them through gossip / publish however setting the private entry flag shouldn't matter where it comes from as we shouldn't be gossiping private store entries anyway. I'll make the change to this.